Adds a new StringHTML Render method

afiune opened the pull request:
This new `StringHTML()` method will allow users to render HTML without
the need of an actual file on disk.

Example:
```go
r.GET("/saludos", func(c *gin.Context) {
	c.StringHTML(200, "<html><body>Hola amigo!</body></html>")
})
```
Signed-off-by: Salim Afiune <afiune@chef.io>

afiune commented:
I added tests so that the coverage doesn't decrease. Now it should be in the same percentage:
```
coverage: 97.0% of statements
```

afiune commented:
@appleboy I'll be interested in having your feedback about this PR. 👍

appleboy commented on context_test.go:
```diff
@@ -733,6 +733,19 @@ func TestContextRenderString(t *testing.T) {
+
+	assert.Equal(t, w.Code, 201)
+	assert.Equal(t, w.Body.String(), "<html><body><h1>test string 2</h1></body><html>")
+	assert.Equal(t, w.HeaderMap.Get("Content-Type"), "text/html; charset=utf-8")
```
```go
assert.Equal(t, 201, w.Code)
assert.Equal(t, "<html><body><h1>test string 2</h1></body><html>", w.Body.String())
assert.Equal(t, "text/html; charset=utf-8", w.HeaderMap.Get("Content-Type"))
```

Equal(t TestingT, `expected`, `actual`)

https://github.com/stretchr/testify/blob/master/assert/assertions.go#L297-L306

appleboy commented on render/render_test.go:
```diff
@@ -161,6 +161,19 @@ func TestRenderString(t *testing.T) {
+
+	assert.NoError(t, err)
+	assert.Equal(t, w.Body.String(), "<html><body><h1>Hola mi amigo numero 1</h1></body></html>")
+	assert.Equal(t, w.Header().Get("Content-Type"), "text/html; charset=utf-8")
```
same as above.

afiune commented:
@appleboy Thanks for the review! 👍  I have updated the assert's to match the right parameter order. Since I took some of the tests as example I thought it would be a good idea to correct them. Finally just added a quick snipped of documentation in the README. Cheers! 🍻

afiune commented:
@appleboy Thanks for the review! 👍 I have updated the assert's to match the right parameter order. Since I took some of the tests as example I thought it would be a good idea to correct them. Finally just added a quick snipped of documentation in the README. Cheers! 🍻

afiune commented:
I just updated my branch to be in sync with `master` 😄

javierprovecho commented:
there was some issue with that same method two years ago here ( https://github.com/gin-gonic/gin/issues/197#issuecomment-102701461 ), the pr is ok, just need to clarify with manu why he didn't wanted that...

sorry for the delay @afiune

afiune commented:
@javierprovecho Thank you for looking out for this change, much appreciated. I read the comment and It make sense to do that if there is not proper method to do it (natively). I just feel it is valuable to have it available and also documented since some micro-services wont be carrying HTML files and instead automatically generating it, so this would be very useful for users that are planning to do so. 

I'll be waiting to hear from Manu. :v:

afiune commented:
@javierprovecho @appleboy Hello! 👋 I was just passing by and remembered this PR, I have rebased from master and updated anything that needs to be updated! 

Is this something we still want to merge?

netroby commented:
+1 need this feature

HarishTeens commented:
Hi, Is this function going to be merged anytime soon?
Or shall I try to make it work?

afiune commented:
It has been a long time, @appleboy @javierprovecho @manucorporat - I am still interested in merging this PR, wondering if anyone is available for a quick review. 👍🏽
